examples/step-7/doc/intro.dox

```diff
@@ -174,6 +150,58 @@ the domain integrals. The details of how this is done are shown in the
 following program.
 
 
+<h3>The method of manufactured solutions</h3>
+
+
+Because we want to verify the convergence of our numerical solution $u_h$,
+we want a setup so that we know the exact solution $u$. This is where
+the Method of Manufactured Solutions comes in: Let us
+choose a function
+@f[
+  \bar u(\mathbf x) =
+  \sum_{i=1}^3 \exp\left(-\frac{|\mathbf x-\mathbf x_i|^2}{\sigma^2}\right)
+@f]
+where the centers $x_i$ of the exponentials are
+  $\mathbf x_1=(-\frac 12,\frac 12)$,
+  $\mathbf x_2=(-\frac 12,-\frac 12)$, and
+  $\mathbf x_3=(\frac 12,-\frac 12)$,
+and the half width is set to $\sigma=\frac {1}{8}$. The method of manufactured
+solution then says: choose
+@f{align*}
+  f &= -\Delta \bar u + \bar u, \\
+  g_1 &= \bar u|_{\Gamma_1}, \\
+  g_2 &= {\mathbf n}\cdot \nabla\bar u|_{\Gamma_2}.
+@f}
+With this particular choice for $f,g_1,g_2$, the solution of the
+original problem must necessarily be $u=\bar u$. In other words, by choosing
+the right hand sides of the equation and the boundary conditions in a
+particular way, we have manufactured ourselves a problem to which we
+know the solution -- a very useful case given that in all but the very
+simplest cases, PDEs do not have solutions we can just write down.
+This then allows us to compute the error of our
+numerical solution. In the code below, we represent $\bar u$ by the
+<code>Solution</code> class, and other classes will be used to
+denote $\bar u|_{\Gamma_1}=g_1$ and ${\mathbf n}\cdot \nabla\bar u|_{\Gamma_2}=g_2$.
+
+@note In principle, you can choose whatever you want for the function $\bar u$
+  above -- here we have simply chosen a sum of three exponentials. In practice,
+  there are two considerations you want to take into account: (i) The function
+  must be simple enough so that you can compute derivatives of the function
+  with not too much effort, for example in order to determine what
+  $f = -\Delta \bar u + \bar u$ is. Since the derivative of an exponential
+  is relatively straightforward to compute, the choice above satisfies this
+  requirement, whereas a function of the kind
+  $\bar u(\mathbf x) = \atan\left(\|\mathbf x\|^{\|\mathbf x\|}\right)$
+  would have presented greater difficulties.
+  (ii) You *don't* want $\bar u$ be a polynomial of low degree. That is
+  because if you choose the polynomial degree of your finite element
+  sufficiently high, you can *exactly* represent this $\bar u$ with
+  the numerical solution $u_h$, making the error zero regardless of
+  how coarse or fine the mesh is. Verifying that this is so is a useful
+  step, but it will not allow you to verify the correct order of
+  convergence of $\|u-u_h\|$ as a function of the mesh size $h$.
```

Comment:
I would also add a third consideration here (this would need an adaptation above as well):
```suggestion
  convergence of $\|u-u_h\|$ as a function of the mesh size $h$ in
  the general case of arbitrary $f$.
  (iii) The typical finite element error estimates assume sufficiently
  smooth solutions, i.e., sufficiently smooth domains, right-hand sides
  $f$ and boundary conditions. A manufactured solution might hide
  geometric complexities, such as singularities in the solution near
  re-entrant corners appearing in real simulations 
  (as considered in the step-27 tutorial program, for example), or trigger
  non-optimal convergence rates by non-smooth functions $f$. Make sure
  to read up on the finite element convergence theory for a more profound
  understanding.
  
```